cmake/therock_detect_python_versions.cmake

```diff
@@ -0,0 +1,89 @@
+# Function to detect available Python versions on the system
```

Comment:
Let's switch to something more declarative vs heuristics. I think this should have a global option like: THEROCK_DIST_PYTHON_EXECUTABLES with a comment like "List of python executables to use for multi-version python builds".

In reality, we will set this to a hard-coded list only when building on manylinux (where we will want a subset of the versions installed in /opt/python-*/bin (we don't want all of them there even and need this to be a configured decision vs auto discovered).

For all other builds, if that variable is not set, default it to the system Python3_EXECUTABLE and only build for one thing.